src/screens/NetworkSettings/CustomNetworkSettingsScreen.js

```diff
@@ -0,0 +1,222 @@
+   */
+  const handleInputChange = (name) => {
+    return (value) => {
+      setFormData({
+        ...formData,
+        [name]: value,
+      });
+    };
+  };
+
+  const handleFeedbackModalDismiss = () => {
+    dispatch(networkSettingsUpdateReady());
+  };
+
+  const handleSubmit = () => {
+    // Validation logic
+    const errors = {};
+
+    if (!formData.nodeUrl) {
+      errors.nodeUrl = t`nodeUrl is required.`;
+    }
+
+    if (!formData.explorerUrl) {
+      errors.explorerUrl = t`explorerUrl is required.`;
+    }
+
+    if (!formData.explorerServiceUrl) {
+      errors.explorerServiceUrl = t`explorerServiceUrl is required.`;
+    }
+
+    setErrorMessages(errors);
+    if (Object.keys(errors).length > 0) {
+      return;
+    }
+
+    dispatch(networkSettingsUpdate(formData));
+  };
+
+  useEffect(() => {
+    setErrorMessages({
+      nodeUrl: networkSettingsErrors?.nodeUrl || '',
+      explorerUrl: networkSettingsErrors?.explorerUrl || '',
+      explorerServiceUrl: networkSettingsErrors?.explorerServiceUrl || '',
+      walletServiceUrl: networkSettingsErrors?.walletServiceUrl || '',
+      walletServiceWsUrl: networkSettingsErrors?.walletServiceWsUrl || '',
+    });
+  }, [networkSettingsErrors]);
+
+  return (
+    <View style={styles.container}>
+      <HathorHeader
+        title={customNetworkSettingsTitleText}
+        onBackPress={() => navigation.goBack()}
+      />
+
+      {isLoading(networkSettingsStatus) && (
+        <FeedbackModal
+          icon={<Spinner />}
+          text={feedbackLoadingText}
+        />
+      )}
+
+      {hasFailed(networkSettingsStatus) && (
+        <FeedbackModal
+          icon={(<Image source={errorIcon} style={styles.feedbackModalIcon} resizeMode='contain' />)}
+          text={feedbackFailedText}
+          onDismiss={handleFeedbackModalDismiss}
+        />
+      )}
+
+      <View style={styles.content}>
+        <View style={styles.warningContainer}>
+          <Text style={styles.warningMessage}>{warningText}</Text>
+        </View>
+        <SimpleInput
+          containerStyle={styles.input}
+          label={t`Node URL`}
+          autoFocus
+          onChangeText={handleInputChange('nodeUrl')}
+          error={errorMessages.nodeUrl}
+          value={formData.nodeUrl}
+        />
+
+        <SimpleInput
+          containerStyle={styles.input}
+          label={t`Explorer URL`}
+          autoFocus
+          onChangeText={handleInputChange('explorerUrl')}
+          error={errorMessages.explorerUrl}
+          value={formData.explorerUrl}
+        />
+
+        <SimpleInput
+          containerStyle={styles.input}
+          label={t`Explorer Service URL`}
+          autoFocus
+          onChangeText={handleInputChange('explorerServiceUrl')}
+          error={errorMessages.explorerServiceUrl}
+          value={formData.explorerServiceUrl}
+        />
+
+        <SimpleInput
+          containerStyle={styles.input}
+          label={t`Wallet Service URL (optional)`}
```

Comment:
It's ok for me to keep it but I think I prefer marking with an `*` the required fields. Writing 'optional' makes the UI less clean, in my opinion.

What do you think?

Reply:
I know that `*` is ubiquitous, however "optional" is more compliant with accessibility aspects.